tools/assets-automation/assets-maintenance-tool/Azure.Sdk.Tools.Assets.MaintenanceTool/Scan/AssetsScanner.cs

```diff
@@ -0,0 +1,359 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Azure.Sdk.Tools.Assets.MaintenanceTool.Model;
+using Azure.Sdk.Tools.TestProxy.Common;
+using Azure.Sdk.Tools.TestProxy.Common.Exceptions;
+using Azure.Sdk.Tools.TestProxy.Store;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace Azure.Sdk.Tools.Assets.MaintenanceTool.Scan
+{
+    /// <summary>
+    /// Used to walk through repo configurations and locate all assets.
+    /// </summary>
+    public class AssetsScanner
```

Comment:
As a general design remark, I see this class is heavily coupled to filesystem IO operations in many places. I wonder if it would be possible to detangle this: have one class whose primary focus is filesystem I/O and another class that is functionally pure and never touches the file system directly. Separation of concerns, hexagonal architecture, "mutable shell + functional core", and all that.